src/client/grpc_client.erl

```diff
@@ -564,6 +615,55 @@ gun_last_reason(Pid) ->
 %%--------------------------------------------------------------------
 %% Helpers
 
+flush_streams(Nowts, State = #state{streams = Streams,
+                                    gun_pid = GunPid,
+                                    client_opts = ClientOpts}) ->
+    Intv = maps:get(stream_batch_delay_ms, ClientOpts, ?DEFAULT_STREAMING_DELAY),
+    NStreams =
+        maps:map(
+          fun(_, Stream = #{sendbuff_size := 0}) ->
+              Stream;
+             (_, Stream = #{sendbuff_last_flush_ts := Ts})
+               when Nowts < (Ts + Intv) ->
+                  Stream;
+             (StreamRef, Stream = #{sendbuff_last_flush_ts := Ts})
+               when Nowts >= (Ts + Intv) ->
+                  flush_data(Nowts, StreamRef, Stream, GunPid)
+          end, Streams),
+   State#state{streams = NStreams}.
+
+maybe_send_data(Bytes, IsFin, StreamRef,
+          Stream = #{st := {_, _RS},
+                     sendbuff := IolistData0,
+                     sendbuff_size := BufferSize0}, GunPid, ClientOpts) ->
```

Comment:
Pass `BatchSizeLimit` as argument instead of `ClientOpts`.
Then we can call `maybe_send_data(..., _BatchSizeLimit = 0)` to force flush all streams.
could be useful in the downgrade code.

Reply:
cool idea!